roles/sap_hana_preconfigure/tasks/SLES/configuration.yml

```diff
@@ -32,6 +32,45 @@
       ansible.builtin.debug:
         var: __sap_hana_preconfigure_fact_solution_configured
 
+- name: Set GRUB entries
+  when: __sap_hana_preconfigure_run_saptune
+  handlers:
+    - name: GRUB_post-update_configuration
+      block:
+        - name: GRUB mkconfig
+          ansible.builtin.shell:
+            cmd: grub2-mkconfig -o /boot/grub2/grub.cfg
+
+        - name: GRUB.cfg permissions
+          ansible.builtin.file:
+            path: /boot/grub2/grub.cfg
+            mode: "0600"
+  block:
+    # Reason for noqa:
+    # no-changed-when: the regex do a check on the element before apply the
+    #   changed item, this prevent a replace to an element that is already in
+    #   the configuration
+    - name: Set GRUB entries # noqa no-changed-when
+      ansible.builtin.lineinfile:
+        path: /etc/default/grub
+        regexp: '^(GRUB_CMDLINE_LINUX_DEFAULT=(?!.* {{ item }}).*). *$'
+        line: "\\1 {{ item }}\""
+      register: set_grub_entries
+      with_items:
+        - "splash=silent"
+        - "mitigations=auto"
+        - "quiet"
+        - "numa_balancing=disable"
+        - "transparent_hugepage=never"
+        - "intel_idle.max_cstate=1"
+        - "processor.max_cstate=1"
+        - "audit=1"
+
+    - name: Trigger grub update if necessary
+      ansible.builtin.meta: noop
```

Comment:
Not sure if you saw this comment. I accidentally had saved it with "Start a review" instead of "Add single comment". Now saving it again, this time using the latter.

I suggest not to use this. The page https://docs.ansible.com/ansible/latest/collections/ansible/builtin/meta_module.html says about `noop`: 
`It is mainly used internally and not recommended for general use.`

Maybe you can use a code similar to task: https://github.com/sap-linuxlab/community.sap_install/blob/7720d7a025a8cca1c7dde23b46df8174e2fdf0fc/roles/sap_general_preconfigure/tasks/RedHat/generic/configure-selinux.yml#L39 ?